astropy/io/votable/converters.py

```diff
@@ -674,19 +674,25 @@ def __init__(self, field, config=None, pos=None):
 
         precision = field.precision
         width = field.width
-        format_parts = ['%']
+
+        if precision is None:
+            format_parts = ['{0']
```

Comment:
I did a test and confirmed that this indeed needs to be fix somemore. Whatever you do, you can use the following as baseline.

General setup:

```python
>>> from astropy.io.votable.converters import Float
>>> from astropy.io.votable.tree import Field
>>> f = Field(None, ID='Test', name='Test', datatype='float',
...           precision=None, width=6)
>>> fp = Float(f)
```

With current `master`:

```python
>>> fp._output_format
'%6r'
>>> fp.output(5.6, None)
'   5.6'
```

With your PR in its current state:

```python
>>> fp._output_format
'{06!r}'
>>> fp.output(5.6, None)
.../astropy/io/votable/converters.py in output(self, value, mask)
    743             if not np.isscalar(value):
    744                 value = value.dtype.type(value)
--> 745             result = self._output_format.format(value)
    746             if result.startswith('array'):
    747                 raise RuntimeError()
IndexError: tuple index out of range
```